src/nvim/window.c

```diff
@@ -3768,8 +3769,18 @@ win_free (
   hash_init(&wp->w_vars->dv_hashtab);
   unref_var_dict(wp->w_vars);
 
-  if (prevwin == wp)
-    prevwin = NULL;
+  {
+    tabpage_T *ttp;
+
+    if (prevwin == wp) {
+      prevwin = NULL;
+    }
+    for (ttp = first_tabpage; ttp != NULL; ttp = ttp->tp_next) {
```

Comment:
Can you use the `FOR_ALL_TABS` macro here?  That avoids duplication of the tab page logic and the extra block just for declaring ttp.